include/abb_librws/rws_interface.h

```diff
@@ -40,294 +40,25 @@
-   * \brief A constructor.
-   *
-   * \param name for the name of the option.
-   * \param description for the description of the option.
-   */
-  RobotWareOptionInfo(const std::string& name, const std::string& description)
-  :
-  name(name),
-  description(description)
-  {}
-
-  /**
-   * \brief The option's name.
-   */
-  std::string name;
-
-  /**
-   * \brief The options's description.
-   */
-  std::string description;
-};
-
-/**
- * \brief A struct for containing information about a RAPID module.
- */
-struct RAPIDModuleInfo
-{
-  /**
-   * \brief A constructor.
-   *
-   * \param name for the name of the module.
-   * \param type for the type of the module.
-   */
-  RAPIDModuleInfo(const std::string& name, const std::string& type)
-  :
-  name(name),
-  type(type)
-  {}
-
-  /**
-   * \brief The module's name.
-   */
-  std::string name;
-
-  /**
-   * \brief The module's type.
-   */
-  std::string type;
-};
-
-/**
- * \brief A struct for containing information about a RAPID task.
- */
-struct RAPIDTaskInfo
-{
-  /**
-   * \brief A constructor.
-   *
-   * \param name for the name of the task.
-   * \param is_motion_task indicating if the task is a motion task or not.
-   * \param is_active indicating if the task is active or not.
-   * \param execution_state indicating the task's current execution state.
-   */
-  RAPIDTaskInfo(const std::string& name,
-                const bool is_motion_task,
-                const bool is_active,
-                const RAPIDTaskExecutionState execution_state)
-  :
-  name(name),
-  is_motion_task(is_motion_task),
-  is_active(is_active),
-  execution_state(execution_state)
-  {}
-
-  /**
-   * \brief The task's name.
-   */
-  std::string name;
-
-  /**
-   * \brief Flag indicating if the task is a motion task.
-   */
-  bool is_motion_task;
-
-  /**
-   * \brief Flag indicating if the task is active or not.
-   */
-  bool is_active;
-
-  /**
-   * \brief The current execution state of the task.
-   */
-  RAPIDTaskExecutionState execution_state;
-};
-
-
-/**
- * \brief A struct for containing static information (at least during runtime) about the robot controller.
- */
-struct StaticInfo
-{
-  /**
-   * \brief Information about the defined RAPID tasks.
-   */
-  std::vector<RAPIDTaskInfo> rapid_tasks;
-
-  /**
-   * \brief System information.
-   */
-  SystemInfo system_info;
-};
-
-
 /**
- * \brief Mapping from IO signal name to a value.
+ * \brief User-friendly interface to Robot Web Services (RWS) 1.0.
  *
- * The value of a digital signal is a \a bool, the value of an analog signal is a \a float.
- */
-using IOSignalInfo = std::map<std::string, std::variant<bool, float>>;
-
-
-/**
- * \brief A class for wrapping a Robot Web Services (RWS) client in a more user friendly interface.
+ * Only a subset of the features available in RWS are implemented here.
+ * See https://developercenter.robotstudio.com/api/rwsApi/ for details about RWS 1.0
  */
 class RWSInterface
+: private MastershipManager
```

Comment:
Is inheritance necessary? Or should we use composition here?
If I understand correctly, we need this inheritance only in `getMastership()` in `Mastership {*this}`, so it should be easy to just create another member variable and pass it there.

Reply:
Inheritance is the simplest solution here, and it doesn't hurt because it is a `private` inheritance (noone outside of the class knows about it).